src/statement.d

```diff
@@ -5149,6 +5154,19 @@ public:
                 }
             }
         }
+
+        if (sc.func)
+        {
+            flags |= sc.func.flags & (FUNCFLAGcppExceptions | FUNCFLAGdExceptions);
+            sc.func.flags |= flags;
+            if (flags == (FUNCFLAGcppExceptions | FUNCFLAGdExceptions))
+            {
+                error("cannot mix catching D and C++ exceptions in the same function");
```

Comment:
I've read it, and still don't understand why you need two personality functions.  There is no technical difficulty in using the same for both.